src/Mremi/UrlShortener/Provider/Bitly/BitlyProvider.php

```diff
@@ -117,28 +128,25 @@ protected function createClient()
     /**
      * Validates the Bit.ly's response and returns it whether the status code is 200.
      *
-     * @param string $apiRawResponse
+     * @param string  $apiRawResponse
+     * @param integer $statusCode
      *
      * @return object
      *
      * @throws InvalidApiResponseException
      */
-    private function validate($apiRawResponse)
+    private function validate($apiRawResponse, $statusCode)
     {
         $response = json_decode($apiRawResponse);
 
         if (null === $response) {
             throw new InvalidApiResponseException('Bit.ly response is probably mal-formed because cannot be json-decoded.');
         }
 
-        if (!property_exists($response, 'status_code')) {
-            throw new InvalidApiResponseException('Property "status_code" does not exist within Bit.ly response.');
-        }
-
-        if (200 !== $response->status_code) {
+        if ($statusCode < 200 || $statusCode >= 300) {
```

Comment:
Why do you change the existing rule (different than 200)?

Reply:
I am the one that proposed this change.

The [API indicates](https://dev.bitly.com/v4/#operation/createBitlink) it could return a 200 or 201 response.